inpars/query_eval.py

```diff
@@ -26,8 +29,8 @@ def __init__(
             else:
                 self.tokenizers[model] = AutoTokenizer.from_pretrained(model)
                 self.models[model] = AutoModel.from_pretrained(model).to(self.device)
-        self.weights = torch.tensor(weights, device=self.device)
-        self.doc_embeddings = None
+        self.weights = OrderedDict({model: weight for model, weight in zip(models, weights)})
+        self.doc_embeddings = OrderedDict()
```

Comment:
Not sure what difference might an OrderedDict make here, considering that [normal dicts are almost the same](https://stackoverflow.com/a/58676384).

Reply:
just making 100% sure nothing gets messed up. Ideally I'd like to get rid of dictionaries completely.